plugins/transfer/plugin.go

```diff
@@ -64,9 +74,16 @@ func init() {
 					return nil, fmt.Errorf("%s: platform configuration %v invalid", plugin.TransferPlugin, uc.platform)
 				}
 
+				s, err := ic.GetByID(plugin.SnapshotPlugin, uc.snapshotter)
```

Comment:
Looks good to me. I think we can ensure the snapshot supports the platform since
the plugin exports supported platforms https://github.com/containerd/containerd/blob/main/plugin/context.go#L76.
But current plugin package api is not friendly to do that right now. It can be a follow-up enhancement.

Reply:
Should we rely on `SnapshotsService` here? then to `getSnapshotter`
https://github.com/containerd/containerd/blob/main/services/snapshots/service.go#L54-L81